cloud/vmware/vmware_vsan_license.py

```diff
@@ -0,0 +1,181 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2016, Kamil Szczygiel <kamil.szczygiel () intel.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: vmware_vsan_license
+short_description: Manage VMware vSAN licenses
+description:
+    - Manage VMware vSAN licenses
+version_added: 2.1
+author: "Kamil Szczygiel (@kamsz)"
+notes:
+    - Tested on vSphere 6.0
+requirements:
+    - "python >= 2.6"
+    - PyVmomi
+options:
+    hostname:
+        description:
+            - The hostname or IP address of the vSphere vCenter API server
+        required: True
+    username:
+        description:
+            - The username of the vSphere vCenter
+        required: True
+        aliases: ['user', 'admin']
+    password:
+        description:
+            - The password of the vSphere vCenter
+        required: True
+        aliases: ['pass', 'pwd']
+    datacenter_name:
+        description:
+            - Name of the datacenter where licensed cluster resides
+    cluster_name:
+        description:
+            - Name of the cluster that will be licensed
+    license_key:
+        description:
+            - License key that will be used for vSAN licensing
+        required: True
+    state:
+        description:
+            - If the datacenter should be present or absent
+        choices: ['present', 'absent']
+        required: True
```

Comment:
in many modules, state is defaulted to 'present'. It has become a convention. If you follow this convention remove the `required: true` and add `default: present`